libmount/src/mountP.h

```diff
@@ -329,6 +335,12 @@ struct libmnt_context
 
 	int	syscall_status;	/* 1: not called yet, 0: success, <0: -errno */
 
+	struct {
+		int fd; /* file descriptor of namespace */
+		struct libmnt_cache *cache; /* paths cache associated with NS */
+	} namespaces[MNT_NS_SIZE];
```

Comment:
Not sure if I like the array, what about to use    
```
struct libmnt_ns  orig_ns, tgt_ns;
```
and add functions mnt_context_get_target_ns() and mnt_context_get_orig_ns() and use it rather than MNT_NS_* enums?

So, within code:
```
anyfunc() {
    struct libmnt_ns *orgns;

    orgns = mnt_context_switch_ns(cxt, mnt_context_get_target_ns(cxt));
    ....
    mnt_context_switch_ns(cxt, orgns);
    return 0;
}
```
IMHO it's more readable and it hides details how thing is implemented. Maybe we can add mnt_context_switch_target_ns() as a shortcut.